HW2/P2/parallel_vector.pyx

```diff
@@ -81,10 +81,16 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           #for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
+                   acquire(&locks[dest[idx]])
```

Comment:
I did this a bit differently, but thinking about it now, it might be better in some cases to really try to make sure you're acquiring and releasing in a 1:1 ratio. Nice!